src/components/ArticleCard.jsx

```diff
@@ -1,19 +1,50 @@
 import React from 'react'
+import {patchArticleVote} from "../API"
+import {useState, useEffect} from "react"
+import CommentCard from "./CommentCard"
 
 const ArticleCard = ({article}) => {
-    return (
+
+    const [voteCount, setVoteCount] = useState(article.votes);
+    const [voteError, setVoteError] = useState("")
+
+    
+    useEffect(() => {
+        setVoteCount(voteCount)
+    }, [voteCount])
+
+    const upVote = (event) => {
+        return patchArticleVote(event.target.value).then(() => {
+         setVoteCount((currCount) => currCount + 1)
+         
+    }).catch((err) => {
+        setVoteError(<p className="error-message">*Vote was not counted, please refresh and try again!*</p>)
+    })
+}
+    const downVote = (event) => {
+        return patchArticleVote(event.target.value, -1).then(() => {
+            setVoteCount((currCount) => currCount - 1)
+            
```

Comment:
I'd like to see optimistic rendering of the vote change. Think about the order of events and what the user sees when they click. Think about how you will deal with the vote NOT updating on the backend